dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/hadoop/BashTask.java

```diff
@@ -53,8 +53,8 @@ public String getTargetPath() {
   private void doRun(ByteSink outputSink, ByteSink errorSink, ByteSink exitStatusSink)
       throws IOException {
     String scriptFilename = scriptName + ".sh";
-    File scriptFile = HadoopScripts.extract(scriptFilename);
-    CommandLine cmdLine = CommandLine.parse("/bin/bash " + scriptFile.getAbsolutePath());
+    Path scriptFile = HadoopScripts.extract(scriptFilename);
+    CommandLine cmdLine = CommandLine.parse("/bin/bash " + scriptFile.toAbsolutePath());
```

Comment:
This looks as if it's parsing the commandline, which might break if the script-file path has a space in it. commons-exec considerably predates the new Java ProcessBuilder interface - is it still worth using commons-exec, or should we be using ProcessBuilder with a pre-split `String[]` (which we basically already have)?

This looks vulnerable to types of injection where java.io.tmpdir="/foo -c /bin/sh" or something, and I'd rather not get into that territory at all.

Reply:
I've removed the dependency on `commons-exec`. I've changed the implementation to `ProcessBuilder` and I pass the arguments individually instead of parsing the whole command line.

I've also removed the code that used the system's temporary directory. The scripts are extracted to the working directory now.